### PR TITLE
feat: NULLS FIRST/LAST clause in ORDER BY

### PR DESCRIPTION
**Pull request by bjouhier**
This PR implements the NULLS FIRST and NULL LAST clauses in ORDER BY. 
https://www.postgresql.org/docs/current/queries-order.html

Example:

```
SELECT a, b FROM ? ORDER BY a ASC NULLS FIRST, b ASC NULLS LAST
```

I encountered 2 problems with `npm run jison`:

- It changes the declaration of `locateNearestErrorRecoveryRule` from a `var` declaration to a `function` declaration. Then `npm test` fails with `GulpUglifyError: unable to minify JavaScript`. I worked around it by fixing the generated file manually.
- After fixing the file, the tests run but one of them fails. This is because some code  (`$1.as = 'default';`) was commented out in alasqlparser.js but not in alasqlparser.json. If fixed this one by commenting it out also in  alasqlparser.json.

Note: these 2 problems are not related to my PR. You can reproduce them on master with `npm run jison && npm test`. I only fixed the second one in this PR

### CODE REVIEW

File: src/426orderby.js [hunk 1]
```diff
@@ -12,9 +12,16 @@ yy.Select.prototype.compileOrder = function(query, params) {
...
 			//			console.log(994, func);
+			var nullsOrder = 
+				this.order[0].nullsOrder == 'FIRST' ? -1 :
+				this.order[0].nullsOrder == 'LAST' ? +1 : 0
 			return function(a, b) {
 				var ra = func(a),
 					rb = func(b);
+				if (nullsOrder) {
+					if (ra === null) return rb === null ? 0 : nullsOrder;
+					if (rb === null) return -nullsOrder;
```
Comment (sdeniaud) [inline src/426orderby.js:23]:
```suggestion
					if (ra == null) return rb == null ? 0 : nullsOrder;
					if (rb == null) return -nullsOrder;
```

**Comment by mathiasrw** [comment]
Awesome!

Looking forward to digg deep into this PR :)

Sorry about the build issues - i met them my self yesterday. We dont always build the JISON partser and an upgrade of the build components broke the build.

**Comment by bjouhier** [comment]
@mathiasrw Thanks. I wanted to submit an issue to jison for the locateNearestErrorRecoveryRule issue and I saw that you already submitted a PR almost 2 years ago. Any hope to get it merged?

**Comment by mathiasrw** [comment]
Its out now as v0.6.0. I really felt inspired by the quality of the contribution. Thank you. 


Regarding `locateNearestErrorRecoveryRule` I really ended up [shaving the yak](https://seths.blog/2005/03/dont_shave_that/) deciding to make what was to be a community fork of jison, work on merging some of the many good PRs to the original repo and then finding out that jison-gho is a thing (active kiiiind of...) and threw it all up in the air and fixing it by just making a simple search and replace in the build.

**Comment by bjouhier** [comment]
Thanks for publishing it quickly. This engine is really cool!

**Pull request merged**